### PR TITLE
Add dedicated "Supported Models" page

### PR DESCRIPTION
**Pull request by chillatom**
Adds a dedicated page under `Cody > Capabilities > supported-models`

Gives a clear layout of which LLMs are supported for chat and completions and for which account types (Free, Pro, Enterprise)

### CODE REVIEW

**Review by arafatkatze — APPROVED** [review]
image

Built the page locally and verified pro/free usage with both of my accounts to confirm that this documentation is 100% correct.


The minor nit is that to add is that in autocomplete you get a set of models available with a provider so for example fireworks would let you choose different models 

image

File: docs/cody/capabilities/supported-models.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+| Anthropic| [claude-3 Opus](https://docs.anthropic.com/claude/docs/models-overview#model-comparison) | - | ✅ | *coming soon* |
+| Mistral| [mixtral 8x7b](https://mistral.ai/technology/#models:~:text=of%20use%20cases.-,Mixtral%208x7B,-Currently%20the%20best) | - | ✅ | - |
+| Ollama* | [variety](https://ollama.com/) | *experimental*| *experimental* | - |
+||||||
+
+## Autocomplete
+
+Cody uses a set of models for autocomplete which are suited for the low latency use case. 
+
+| **Provider** | **Model** | **Free** | **Pro** | **Enterprise** |
```
Comment (arafatkatze) [inline docs/cody/capabilities/supported-models.mdx:25]: Maybe change _Model_ to  to _Model set_ or _Models_ 
[See this](https://github.com/sourcegraph/docs/pull/217#pullrequestreview-1965750801)

File: docs/cody/capabilities/supported-models.mdx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Cody supports a variety of cutting edge large language models for use in Chat, allowing you to select the best model for your use case.  Free users are defaulted to Claude 2.0 model from Anthropic, while Pro users have access to select any supported model.
+
+| **Provider** | **Model** | **Free** | **Pro** | **Enterprise** |
+|:--|:--|:--|:--|:--|
+| OpenAI | [gpt-3.5 turbo](https://platform.openai.com/docs/models/gpt-3-5-turbo) | - | ✅ | ✅ |
+| OpenAI | [gpt-4 turbo](https://platform.openai.com/docs/models/gpt-4-and-gpt-4-turbo#:~:text=TRAINING%20DATA-,gpt%2D4%2D0125%2Dpreview,-New%20GPT%2D4) | - | ✅ | ✅ |
+| Anthropic| [claude Instant](https://docs.anthropic.com/claude/docs/models-overview#model-comparison) | - | ✅ | ✅ |
+| Anthropic| [claude-2.0](https://docs.anthropic.com/claude/docs/models-overview#model-comparison) | ✅ | ✅ | ✅ |
+| Anthropic| [claude-2.1](https://docs.anthropic.com/claude/docs/models-overview#model-comparison) | - | ✅ | ✅ |
+| Anthropic| [claude-3 Haiku](https://docs.anthropic.com/claude/docs/models-overview#model-comparison) | - | ✅ | *coming soon* |
```
Comment (arafatkatze) [inline docs/cody/capabilities/supported-models.mdx:14]: We are changing default chat for both to be [sonnet](https://github.com/sourcegraph/cody/pull/3575) in the latest pr that you co-ordinated with philipp. 

image

Comment (arafatkatze) [inline docs/cody/capabilities/supported-models.mdx:14]: We are changing default chat for both to be [sonnet](https://github.com/sourcegraph/cody/pull/3575) in the latest pr that you co-ordinated with philipp. 

image

Comment (chillatom) [reply docs/cody/capabilities/supported-models.mdx:14]: Yeah, I just wanted this PR to map to current state.  I'll update as soon as we roll out Sonnet as default.

Comment (arafatkatze) [reply docs/cody/capabilities/supported-models.mdx:14]: Cool cool

**Pull request merged**